circuits/kimchi/src/gates/generic.rs

```diff
@@ -27,27 +27,19 @@ impl<F: FftField> CircuitGate<F> {
     }
 
     /// creates an addition gate
-    pub fn create_generic_add(row: usize, wires: GateWires) -> Self {
+    pub fn create_generic_add(row: usize, wires: GateWires, left_coeff: F, right_coeff: F) -> Self {
         let on = F::one();
         let off = F::zero();
-        let qw: [F; GENERICS] = array_init(|col| {
-            if col < 2 {
-                on
-            } else if col == 2 {
-                -on
-            } else {
-                off
-            }
-        });
+        let qw: [F; GENERICS] = [left_coeff, right_coeff, -on];
         Self::create_generic(row, wires, qw, off, off)
     }
 
     /// creates a multiplication gate
     pub fn create_generic_mul(row: usize, wires: GateWires) -> Self {
         let on = F::one();
         let off = F::zero();
-        let qw: [F; GENERICS] = array_init(|col| if col == 2 { on } else { off });
-        Self::create_generic(row, wires, qw, -on, off)
+        let qw: [F; GENERICS] = [off, off, -on];
+        Self::create_generic(row, wires, qw, on, off)
```

Comment:
cleaner to have a negation of the output selector coeff, instead of doing it on the multiplication selector coeff